packages/wonder-blocks-modal/components/modal-launcher/index.test.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+import React from "react";
+import {shallow} from "enzyme";
+
+import ModalLauncher from "./index.js";
+
+describe("ModalLauncher", () => {
+    test("Children can launch the modal", () => {
+        const wrapper = shallow(
+            <ModalLauncher modal={<div />}>
+                {({openModal}) => <button onClick={openModal} />}
+            </ModalLauncher>,
+        );
+        expect(wrapper.find("ModalLauncherPortal")).toHaveLength(0);
+        wrapper.find("button").simulate("click");
+        expect(wrapper.find("ModalLauncherPortal")).toHaveLength(1);
+    });
+
+    test("Modal can close itself after launching", (done) => {
+        let opened = false;
+
+        // Once the modal mounts, we'll immediately self-close it on the next
+        // tick, to test the children's ability to self-close.
+        //
+        // NOTE(mdr): It would be nice to have this be, like, a close button
+        //     that closes when you click it. But that requires the button to
+        //     actually _mount_, which means we'd need to do a full DOM render
+        //     including `ModalLauncherPortal`, and that seems to be going
+        //     beyond the scope of this test. Really we just want to check that
+        //     this function receives a `closeModal` argument that works.
+        const modalFn = ({closeModal}) => {
+            expect(opened).toBe(true);
+            setImmediate(closeModal);
```

Comment:
Can you add a comment about why `setImmediate` is necessary here?

Reply:
Hmm good call. Mostly I'm trying to simulate more realistic behavior: `closeModal` really shouldn't be called immediately during this render phase, and in fact will trigger a complaint from React that you're calling `setState` during `render`. (Hmm, maybe I should actually explicitly disallow this, and write a test for it!)

Reply:
Eh, I guess the consequences of calling `setState` during `render` aren't so bad, and the error will announce itself if a dev makes that mistake. We shouldn't do it in our test, and I'll explain that (and maybe leave a comment), but explicitly disallowing it is probably excessive.